### DISCUSSION TITLE
Curly brackets and normal brackets when calling rsx macro.

### DISCUSSION THREAD

**Discussion by onweru** [DioxusLabs/dioxus#2731] [Q&A] [answered]
can `{` & `(` be used interchangeably when calling `rsx` macro?

```rust
fn Component -> Element {
  rsx! (
    div { "some el" } 
  )
}

// or

fn Component -> Element {
 rsx! {
    div { "some el" } 
 }
}
```

Would any of that be responsible for this log by dioxus logger?

> It looks like you called the component Field like a function instead of a component. Components should be called with braces like `Field { prop: value }` instead of as a function

**Comment by ealmloff** [comment]
Yes, macros can be called with `{}`'s, `[]`'s, or `()`'s. `rsx!{}`, `rsx!()`, and `rsx![]` all do exactly the same thing

> Would any of that be responsible for this log by dioxus logger?

No, but you could try bumping dioxus. Dioxus was throwing that warning in a few situations that are fine. That issue was fixed in https://github.com/DioxusLabs/dioxus/pull/2649

**Reply by onweru** [reply]
Thank you for clarifying that. I didn't know we could use `rsx![]`. 

I'm tapping the latest commit on main. I still see it.

```
dioxus = { git = "https://github.com/DioxusLabs/dioxus", rev = "3db24934d6f2ef074e58eb9b116228327a963c9f", features=["web", "router"] }
```

**Comment by ealmloff** [comment]
> I'm tapping the latest commit on main. I still see it.

Then it could be an error. Do you have anything like `Field()` in your code?

**Reply by onweru** [reply]
I have a `Field` struct

```rust
#[derive(Clone, Debug, Default, Deserialize, PartialEq)]
pub struct Field {
  pub id: String,
  pub value: String,
}
```

Other than that, I have components that end with Field e.g `RadioField` or `UploadField`

**Reply by onweru** [reply]
Update: turns I had a component called `Field`. At first I thought _Field_ was sort of a reserved name, I was wrong.

I was importing that component as follows, to avoid conflicts with the Field struct mentioned before
```rust
use crate::components::Field as TextField

rsx! {
  TextField { }
}
```

I renamed the struct. After that, changing that import to 👇🏼 made the error go away. 

```rust
use crate::components::Field

rsx! {
  Field { }
}

```

**Comment by ealmloff** [comment, marked as answer]
That is a bug. Dioxus isn't seeing through the re-export for the type name properly when triggering the lint

Your code is fine and it is safe to ignore the warning

**Reply by ealmloff** [reply]
https://github.com/DioxusLabs/dioxus/issues/2743
